### ISSUE TITLE
Screengrab:  output files "couldn't be opened because [I] don't have permission"

### ISSUE THREAD

**Issue by dpettit** [fastlane/fastlane#8322]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
Screengrab produces snapshots, but the resulting .png files can't be opened because I don't have the appropriate permissions.  Screengrab is creating the "metadata/android/en-US/images/phoneScreenshots /sevenInchScreenshots folders and snapshots, so I'm not sure how I am supposed to set things up ahead of time such that I will have the permissions.  Is there a certain user account that screengrab uses that I can grant permissions to, or what is the best approach to solving this?

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

[INSERT OUTPUT HERE]
n/a

### Environment
Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

[INSERT OUTPUT HERE]
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.12.3                                     |
| Ruby                | 2.3.3                                       |
| Bundler?            | false                                       |
| Git                 | git version 2.10.1 (Apple Git-78)           |
| Installation Source | ~/.rbenv/versions/2.3.3/bin/fastlane        |
| Host                | Mac OS X 10.12.3 (16D32)                    |
| Ruby Lib Dir        | ~/.rbenv/versions/2.3.3/lib                 |
| OpenSSL Version     | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained        | false                                       |
| Is homebrew         | false                                       |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.18.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 1.5.1        |
| i18n                      | 0.7.0        |
| json                      | 1.8.3        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| minitest                  | 5.8.5        |
| activesupport             | 4.2.7.1      |
| CFPropertyList            | 2.3.4        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.2        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.2        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.4        |
| addressable               | 2.5.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| highline                  | 1.7.8        |
| commander                 | 4.4.3        |
| excon                     | 0.54.0       |
| faraday                   | 0.10.1       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.0.1        |
| gh_inspector              | 1.0.2        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.20       |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| bundler                   | 1.13.7       |
| faraday_middleware        | 0.10.1       |
| io-console                | 0.4.5        |
</details>

*generated on:* **2017-02-21**
</details>

**Comment by dpettit** [comment]
This behavior doesn't happen all of the time; it is very hit and miss.

**Comment by mfurtak** [comment]
I can't think of a reason this would be, and it's even stranger that it should be inconsistent!

When it happens, can you send me the output of `ls -lha` on the directory with the PNGs in it? What user/group are they winding up owned as?

**Comment by dpettit** [comment]
Thank you for the quick response!  You guys are amazing!

I was able to resolve it for now (at least, I'm keeping my fingers crossed).  But I will let you know if it comes up again.  For now, I'm just going to close this as an issue, because I found something that appears to be working.

The way I resolved it was by doing the following:
Back up all data. Quit Preview if it's running.
 
Hold down the shift key and select Go ▹ Library from the Finder menu bar. From the Library folder, delete the following items, if they exist:
 
     Caches/com.apple.Preview
     Containers/com.apple.Preview
     Preferences/com.apple.Preview.LSSharedFileList.plist
     Preferences/com.apple.Preview.SandboxedPersistentURLs.LSSharedFileList.plist
     Saved Application State/com.apple.Preview.savedState
     Restart machine, and see if things work properly.

As it is, it appears that this may be more of an iMac OS issue, but I wanted to mention it here in case anyone else encounters a similar problem.
